src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/scanOrchestrator.hpp

```diff
@@ -128,46 +143,139 @@ class TScanOrchestrator final : public TOSPrimitives
 
         // Set the retrieved manager name in the global data object
         GlobalData::instance().managerName(managerName);
+        initEventDelayedDispatcher();
     }
     ~TScanOrchestrator() = default;
     // LCOV_EXCL_STOP
 
+    /**
+     * @brief Start the delayed event dispatcher
+     */
+    void initEventDelayedDispatcher()
+    {
+        m_eventDelayedDispatcher = std::make_shared<EventDelayedDispatcher>(
+            // coverity[copy_constructor_call]
+            [this](rocksdb::PinnableSlice& element)
+            {
+                try
+                {
+                    processEvent(element, true);
+                }
+                catch (const WdbDataException& e)
+                {
+                    m_eventDelayedDispatcher->postpone(e.agentId(), std::chrono::seconds(DELAYED_POSTPONE_SECONDS));
+                    throw std::runtime_error(e.what());
```

Comment:
NitpicK: We could add the agentId to improve the logging down the line
```suggestion
                    throw std::runtime_error(std::string(e.what()) + " (Agent: " + e.agentId() + ")");
```